hazelcast/src/main/java/com/hazelcast/map/impl/querycache/subscriber/QueryCacheEndToEndProvider.java

```diff
@@ -48,36 +49,40 @@ public QueryCacheEndToEndProvider(ContextMutexFactory mutexFactory) {
         this.mapNameToQueryCaches = new ConcurrentHashMap<String, ConcurrentMap<String, InternalQueryCache<K, V>>>();
     }
 
-    public InternalQueryCache<K, V> getOrCreateQueryCache(String mapName, String cacheId,
+    public InternalQueryCache<K, V> getOrCreateQueryCache(String mapName, String cacheName,
                                                           ConstructorFunction<String, InternalQueryCache<K, V>> constructor) {
         ContextMutexFactory.Mutex mutex = mutexFactory.mutexFor(mapName);
         try {
             synchronized (mutex) {
-                ConcurrentMap<String, InternalQueryCache<K, V>> cacheIdToQueryCache
+                ConcurrentMap<String, InternalQueryCache<K, V>> cacheNameToQueryCache
                         = getOrPutIfAbsent(mapNameToQueryCaches, mapName, ctor);
-                InternalQueryCache<K, V> queryCache = cacheIdToQueryCache.get(cacheId);
-                if (queryCache == null) {
-                    queryCache = constructor.createNew(cacheId);
-                    if (queryCache == NULL_QUERY_CACHE) {
-                        queryCache = null;
-                    } else {
-                        cacheIdToQueryCache.put(cacheId, queryCache);
-                    }
+
+                InternalQueryCache<K, V> queryCache = cacheNameToQueryCache.get(cacheName);
```

Comment:
The check whether the query cache is already available can be executed first, to avoid construction & synchronization on the `mutex` in case the query cache has already been constructed. I think it's worth the small duplication of code.

Reply:
done